### PR TITLE
luci-app-ser2net: created

### PR DESCRIPTION
**Pull request by TimelessNL**
This changeset will add support for ser2net in luci-app-ser2net.

_Note: This depends on https://github.com/openwrt/packages/pull/5302 to be accepted first._

Note2: This changeset is based on https://github.com/JiapengLi/OpenWrt-luci-app-ser2net/blob/master/openwrt-packages-add-luci-app-ser2net-support.patch

### CODE REVIEW

File: applications/luci-app-ser2net/Makefile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#
+# Copyright (C) 2008-2017 The LuCI Team <luci@lists.subsignal.org>
```
Comment (feckert) [inline applications/luci-app-ser2net/Makefile]: As I know please change the Copyright notice to
> Copyright (C) 2018 Jasper Scholte <NightNL@outlook.com>

File: applications/luci-app-ser2net/luasrc/controller/ser2net.lua [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module("luci.controller.ser2net", package.seeall)
```
Comment (feckert) [inline applications/luci-app-ser2net/luasrc/controller/ser2net.lua]: Could you please add the following header on top off the file

> -- Copyright (C) 2018 Jasper Scholte NightNL@outlook.com
> -- Licensed to the public under the Apache License 2.0.

File: applications/luci-app-ser2net/luasrc/model/cbi/ser2net.lua [hunk 3]
```diff
@@ -0,0 +1,58 @@
+m = Map("ser2net", translate("ser2net"),
```
Comment (feckert) [inline applications/luci-app-ser2net/luasrc/model/cbi/ser2net.lua]: Could you please add the following header on top off the file

> -- Copyright (C) 2018 Jasper Scholte NightNL@outlook.com
> -- Licensed to the public under the Apache License 2.0.

File: applications/luci-app-ser2net/luasrc/model/cbi/ser2net.lua [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        translate("The ser2net allows telnet and tcp sessions to be established with a unit's serial ports.<br/>"))
+
+function m.on_after_commit(self)
+        luci.sys.call("/etc/init.d/ser2net enable 1\>/dev/null 2\>/dev/null")
```
Comment (feckert) [inline applications/luci-app-ser2net/luasrc/model/cbi/ser2net.lua]: You could use
> luci.sys.init.enable("ser2net")

File: applications/luci-app-ser2net/luasrc/model/cbi/ser2net.lua [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        luci.sys.call("/etc/init.d/ser2net restart 1\>/dev/null 2\>/dev/null")
```
Comment (feckert) [inline applications/luci-app-ser2net/luasrc/model/cbi/ser2net.lua]: You sould use
 > luci.sys.call("/etc/init.d/ser2net restart >/dev/null 2>&1")

**Comment by arturioxas** [comment]
Changes you submitted seem to write to configuration that only luci understands. In my opinion it seems unreliable, because:
1. ser2net by default uses /etc/ser2net.conf. Therefore initializing won't do anything by default because setting are setting in uci format in uci compatible file
2. changing /etc/ser2net.conf or using example setting won't be reflected in frontend application

**Comment by TimelessNL** [comment]
True. I forgot about this request. The implementation has changed in OpenWrt. So I have to refactor this pullrequest. I'll close it for now.

**Pull request closed without merging**